torch/jit/__init__.py

```diff
@@ -720,6 +758,29 @@ def define(self, lang):
         rcb = createResolutionCallback()
         self._define(lang, rcb, True)
 
+    def _check_modules_are_owned(self, v):
+        if isinstance(v, ScriptModule) and not self._is_script_submodule(v):
+            raise RuntimeError("torch.jit.Const includes a module '{}'".format(type(v).__name__) +
+                               " which is not a submodule of this module")
```

Comment:
Can't we just disallow assigning scripted modules as const variables? This seems to be a lot simpler, and I don't think we get a lot of value from what we have here (since it had to be assigned before anyway)